### DISCUSSION TITLE
How to run a testnet node

### DISCUSSION THREAD

**Discussion by chriswsq** [hiero-ledger/hiero-mirror-node#2573] [Help] [answered]
How to use a server to run a testnet locally, and store data locally without awk and google

**Comment by steven-sheehy** [comment, marked as answer]
The mirror node requires AWS and Google Cloud to retrieve data the main nodes write. If you're wanting to run a main node, you can do so locally for testing purposes using the instructions in the hedera-services [README](https://github.com/hashgraph/hedera-services/blob/master/README.md). You cannot run a Hedera main node for testnet or mainnet since it is a permissioned network.
